pkg/resource/scaling_policy/testdata/test_suite.yaml

```diff
@@ -1,17 +1,115 @@
+         latest_state: "scaling_policy/v1alpha1/create/observed/invalid_create_attempted.yaml"
+         error: "InvalidParameterValue: The service namespace must be one of the following: appstream | cassandra | comprehend | custom-resource | dynamodb | ec2 | ecs | elasticmapreduce | kafka | lambda | rds | sagemaker.\n\tstatus code: 0, request id: "
+     - name: "Put=ValidInput"
+       description: "Put a new ScalingPolicy successfully."
+       given:
+         desired_state: "scaling_policy/v1alpha1/create/desired/valid_before_create.yaml"
+         svc_api:
+           - operation: PutScalingPolicyWithContext
+             output_fixture: "scaling_policy/sdkapi/create_success.yaml"
+       invoke: Create
+       expect:
+         latest_state: "scaling_policy/v1alpha1/create/observed/valid_created.yaml"
+         error: nil
+  - name: "ScalingPolicy delete tests"
+    description: "ScalingPolicy delete tests"
+    scenarios:
+      - name: "Delete=Success"
+        description: "Successfully delete a ScalingPolicy"
+        given:
+          desired_state: "scaling_policy/v1alpha1/create/observed/valid_created.yaml"
+          svc_api:
+            - operation: DeleteScalingPolicyWithContext
+        invoke: Delete
+        expect:
+          error: nil
+      - name: "Delete=Fail"
+        description: "If deletion fails, an error should be thrown"
+        given:
+          desired_state: "scaling_policy/v1alpha1/create/observed/valid_created.yaml"
+          svc_api:
+            - operation: DeleteScalingPolicyWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: Delete
+        expect:
+          latest_state: "scaling_policy/v1alpha1/delete/observed/error_on_delete.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+  - name: "ScalingPolicy readone tests"
+    description: "ScalingPolicy readone tests"
+    scenarios:
+    - name: "ReadOne=MissingRequiredField"
+      description: "Testing ReadOne when required field is missing. No API call is made and returns error."
+      given: 
+        desired_state: "scaling_policy/v1alpha1/readone/desired/missing_field.yaml"
+        svc_api:
+          - operation: DescribeScalingPoliciesWithContext
+            error:
+              message: "resource not found"
+      invoke: ReadOne
+      expect:
+        error: ": resource not found\n\tstatus code: 0, request id: "
+    - name: "ReadOne=NotFound"
+      description: "Testing ReadOne when Describe fails to find the resource on SageMaker"
+      given: 
+        desired_state: "scaling_policy/v1alpha1/create/observed/valid_created.yaml"
+        svc_api:
+          - operation: DescribeScalingPoliciesWithContext
+            error:
+              message: "resource not found"
+      invoke: ReadOne
+      expect:
+        error: ": resource not found\n\tstatus code: 0, request id: "
```

Comment:
what is the difference between this test and the `ReadOne=MissingRequiredField` test ?

Reply:
The `ReadOne=MissingRequiredField` is basically checking that an invalid spec gets caught right away. The `NotFound` checks that if the describe command returns a `notFound` Error, readOne reflects it correctly - the scenario being that the resource does not exist on SageMaker.